### ISSUE TITLE
Build fails with autoconf 2.70

### ISSUE THREAD

**Issue by sandrotosi** [transmission/transmission#1573]
Hello, it was reported to debian (as http://bugs.debian.org/978911) that transmission fails to build using autoconf 2.70; here is the bug report text (it contains some debian references but the failure is shown with a traceback):

```
The package fails to build in a test rebuild on at least amd64 with
autoconf 2.70, but succeeds to build with autoconf 2.69. The
severity of this report will be raised before the bookworm release,
so nothing has to be done for the bullseye release.

The full build log can be found at:
http://qa-logs.debian.net/2020/09/26.ac270/transmission_3.00-1_unstable_ac270.log
The last lines of the build log are at the end of this report.

To build with autoconf 2.70, please install the autoconf package from
experimental:  apt-get -t=experimental install autoconf

[...]
m4/acx-pthread.m4:13: ACX_PTHREAD is expanded from...
configure.ac:116: the top level
configure.ac:116: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:116: You should run autoupdate.
./lib/autoconf/general.m4:2816: AC_TRY_LINK is expanded from...
m4/acx-pthread.m4:13: ACX_PTHREAD is expanded from...
configure.ac:116: the top level
configure.ac:116: warning: The macro `AC_LANG_RESTORE' is obsolete.
configure.ac:116: You should run autoupdate.
./lib/autoconf/lang.m4:135: AC_LANG_RESTORE is expanded from...
m4/acx-pthread.m4:13: ACX_PTHREAD is expanded from...
configure.ac:116: the top level
configure.ac:565: warning: The macro `GLIB_GNU_GETTEXT' is obsolete.
configure.ac:565: You should run autoupdate.
/usr/share/aclocal/glib-gettext.m4:366: GLIB_GNU_GETTEXT is expanded from...
/usr/share/aclocal/glib-gettext.m4:466: AM_GLIB_GNU_GETTEXT is expanded from...
configure.ac:565: the top level
configure.ac:565: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:565: You should run autoupdate.
./lib/autoconf/general.m4:2816: AC_TRY_LINK is expanded from...
lib/m4sugar/m4sh.m4:673: AS_IF is expanded from...
./lib/autoconf/general.m4:2149: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2170: AC_CACHE_CHECK is expanded from...
/usr/share/aclocal/glib-gettext.m4:51: GLIB_LC_MESSAGES is expanded from...
/usr/share/aclocal/glib-gettext.m4:366: GLIB_GNU_GETTEXT is expanded from...
/usr/share/aclocal/glib-gettext.m4:466: AM_GLIB_GNU_GETTEXT is expanded from...
configure.ac:565: the top level
configure.ac:565: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:565: You should run autoupdate.
./lib/autoconf/general.m4:2816: AC_TRY_LINK is expanded from...
lib/m4sugar/m4sh.m4:673: AS_IF is expanded from...
./lib/autoconf/general.m4:2149: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2170: AC_CACHE_CHECK is expanded from...
lib/m4sugar/m4sh.m4:673: AS_IF is expanded from...
./lib/autoconf/headers.m4:89: _AC_CHECK_HEADER_COMPILE is expanded from...
./lib/autoconf/headers.m4:56: AC_CHECK_HEADER is expanded from...
/usr/share/aclocal/glib-gettext.m4:150: GLIB_WITH_NLS is expanded from...
/usr/share/aclocal/glib-gettext.m4:366: GLIB_GNU_GETTEXT is expanded from...
/usr/share/aclocal/glib-gettext.m4:466: AM_GLIB_GNU_GETTEXT is expanded from...
configure.ac:565: the top level
configure.ac:565: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:565: You should run autoupdate.
./lib/autoconf/general.m4:2816: AC_TRY_LINK is expanded from...
lib/m4sugar/m4sh.m4:673: AS_IF is expanded from...
./lib/autoconf/headers.m4:89: _AC_CHECK_HEADER_COMPILE is expanded from...
./lib/autoconf/headers.m4:56: AC_CHECK_HEADER is expanded from...
/usr/share/aclocal/glib-gettext.m4:150: GLIB_WITH_NLS is expanded from...
/usr/share/aclocal/glib-gettext.m4:366: GLIB_GNU_GETTEXT is expanded from...
/usr/share/aclocal/glib-gettext.m4:466: AM_GLIB_GNU_GETTEXT is expanded from...
configure.ac:565: the top level
configure.ac:565: warning: The macro `AC_OUTPUT_COMMANDS' is obsolete.
configure.ac:565: You should run autoupdate.
./lib/autoconf/status.m4:1025: AC_OUTPUT_COMMANDS is expanded from...
/usr/share/aclocal/glib-gettext.m4:150: GLIB_WITH_NLS is expanded from...
/usr/share/aclocal/glib-gettext.m4:366: GLIB_GNU_GETTEXT is expanded from...
/usr/share/aclocal/glib-gettext.m4:466: AM_GLIB_GNU_GETTEXT is expanded from...
configure.ac:565: the top level
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
ERROR: 'IT_PROG_INTLTOOL' must appear in configure.ac for intltool to work.
autoreconf: error: intltoolize failed with exit status: 1
dh_autoreconf: error: autoreconf -f -i returned exit code 1
make: *** [debian/rules:20: build] Error 25
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
```

**Comment by mikedld** [comment]
Starting with 4c7153fa48d86627af3a0ec4e1369632c81484d0, only CMake build system is supported.

**Comment by mikedld** [comment]
May want to fix this if 3.0x ever sees the light, otherwise you should really consider looking into building with CMake for 3.1x (and even for 3.00).

**Comment by sandrotosi** [comment]
sure thing, once 3.1 is out i'll use CMake; the inability to use autoconf 2.70 is not something that will impact Debian in the short term anyway (as it's planned to be introduced only months away from now)  so this bug was mostly a headsup

**Comment by ffontaine** [comment]
It should be noted that this build failure is raised because `AC_PROG_INTLTOOL` stopped working if there is a 'tab' on the beginning of the line since autoconf 2.71: https://www.mail-archive.com/bug-autoconf@gnu.org/msg04851.html.

**Comment by ckerr** [comment]
I'm going to close this issue since Transmission no longer uses autotools. If the build is still failing under CMake, please file a new issue. Thank you!

**Comment by sandrotosi** [comment]
then a release with a fully working CMake is in order, which i believe only happened after 2.70

**Comment by ckerr** [comment]
> then a release with a fully working CMake is in order, which i believe only happened after 3.00 (released 1.5 years ago)

then a release with a fully working CMake is in order, which i believe only happened after 3.00 (released 1.5 years ago)
